atlas/AtlasMl/atlasml/config.py

```
@@ -24,36 +20,116 @@ class Settings(BaseModel):
     def get_api_keys(cls):
+        logger.debug(f"Getting API keys: {cls.get_settings().api_keys}")
         return cls.get_settings().api_keys
 
-settings = Settings.get_settings()
\ No newline at end of file
+
+# Lazy initialization of settings - will be created when first accessed
+_settings = None
+
+
+def get_settings(use_defaults: bool = None) -> Settings:
```

Ruff (0.12.2)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)